### DISCUSSION TITLE
Parameterized multi-row integer inserts with executeQuery

### DISCUSSION THREAD

**Discussion by Medvedscak** [prisma/prisma#15717] [Q&A] [answered]
Hello, I'm new to Prisma and my question is very similar to https://github.com/prisma/prisma/discussions/4177 but I'm trying to insert integers to the table and the solution provided there doesn't work for me(

The dest table contains 2 columns with integers.
To simplify I hardcoded the inserted data:

```
const testArr = ['(1,2)', '(2,2)']

await this.prisma.$executeRaw`
  INSERT INTO public."_CategoryToItem" ("B", "A")
  VALUES ${testArr.join(',')}
  ON CONFLICT DO NOTHING
;`

```

And this is the exception I see (with debug logs attached):

```
INSERT INTO public."_CategoryToItem" ("B", "A")
      VALUES $1
      ON CONFLICT DO NOTHING
  ; ["(1,2),(2,2)"]
Duration: 0ms
[Nest] 1226  - 10/08/2022, 11:41:45 AM   ERROR [ExceptionsHandler] 
Invalid `prisma.$executeRaw()` invocation:

Raw query failed. Code: `42601`. Message: `db error: ERROR: syntax error at or near "$1"`
```

As far as I understand, the issue is that parentheses are added around inserted values (or maybe not).
Please advise the workaround.
Note: I'm using raw query because of working with unsupported type in other related queries so I need the help with the raw query.

UPD: if I use an array of numbers and add only 1 row with Prisma.join() - it works fine:

```
const testArr = [1,3]
return await this.prisma.$executeRaw`
  INSERT INTO public."_CategoryToItem" ("B", "A")
  VALUES (${Prisma.join(testArr)})
  ON CONFLICT DO NOTHING
;
```
But I still can't figure out why I can't just add a template string there... Or which workaround to use (pairs of braces are the blocker)

UPD2: Also checked that it works with the **executeRawUnsafe** but I'd like not to use unsafe options:
```
const testArr = ['(1,6)', '(1,7)']
return await this.prisma.$executeRawUnsafe(
  `INSERT INTO public."_CategoryToItem" ("B", "A")
  VALUES ${testArr.join(',')}
  ON CONFLICT DO NOTHING
;`)
```

**Comment by pantharshit00** [comment, marked as answer]
Hello @Medvedscak,

You can do the following here:
```ts
import { PrismaClient, Prisma } from '@prisma/client'

// ....
// ....
const testArr = [Prisma.sql`(1,2)`, Prisma.sql`(2,2)`]

await prisma.$executeRaw`
  INSERT INTO public."_CategoryToItem" ("B", "A")
  VALUES ${Prisma.join(testArr)}
  ON CONFLICT DO NOTHING
;`
```

Let me know if this worked for you. We use this library under the hood for sql template support so you can reference that: https://github.com/blakeembrey/sql-template-tag

**Reply by Medvedscak** [reply]
Thanks a lot! It helped me. Waiting for the PostGis support though (that's the only reason I have to use raw queries)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
